src/components/workspace/collabEditing/helper.ts

```diff
@@ -38,3 +38,27 @@ export function createNewSession(onSessionCreated: (sessionId: string) => void)
   xmlhttp.open('GET', 'https://' + LINKS.SHAREDB_SERVER + 'gists/latest/', true);
   xmlhttp.send();
 }
+
+export function checkConnnectionAlive(
+  editorSessionId: string,
+  connectionOK: () => void,
+  sessionIdNotFound: () => void,
+  cannotReachServer: () => void
+) {
+  const xmlhttp = new XMLHttpRequest();
+  xmlhttp.onreadystatechange = () => {
+    if (xmlhttp.readyState === 4 && xmlhttp.status === 200) {
+      const state = JSON.parse(xmlhttp.responseText).state;
+      if (state !== true) {

```

Comment:
You have too many if/else branches here. I believe this can be refactored to be more clean?

Try to make the happy path prominent!

Reply:
I can't think there's a way to make it cleaner as the state can only be seen after we get the response.